### PR TITLE
adapt port validation regex

### PR DESCRIPTION
**Pull request by MaximilianMeister**
tested with http://rubular.com/

### CODE REVIEW

**Comment by Itxaka** [comment]
Tested the regex and works perfectly, thanks!
+1

File: lib/crowbar/client/mixin/database.rb [hunk 1]
```diff
@@ -30,7 +30,7 @@ module Database
         REGEX_USERNAME = "(?=^.{4,63}$)(?=^[a-zA-Z0-9_]*$)".freeze
         REGEX_PASSWORD = "(?=^.{4,63}$)(?=^[a-zA-Z0-9_]*$)(?=[a-zA-Z0-9_$&+,:;=?@#|'<>.^*()%!-]*$)".freeze
         REGEX_DATABASE = "(?=^.{1,63}$)(?=^[a-zA-Z0-9_]*$)(?=[a-zA-Z0-9_$&+,:;=?@#|'<>.^*()%!-]*$)".freeze
-        REGEX_PORT = "(?=^.{1,5}$)(?=^[0-9]*$)".freeze
+        REGEX_PORT = "(?=^.{1,5}$)(?=^[0-9]*$)(?=^([1-9][0-9]{0,3}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])$)".freeze
```
Comment (nicolasbock) [inline lib/crowbar/client/mixin/database.rb]: Wouldn't it be easier to convert to `int` and then do a value comparison?

**Comment by MaximilianMeister** [comment]
> Wouldn't it be easier to convert to int and then do a value comparison?

@nicolasbock good idea, that would work too, but AFAIK the UI also works with regex validation, plus the API documentation needs to display a validation regex, so we need it in a few places. and the important bit here is that it always works. are you fine keeping it the way it is to be consistent?

**Comment by nicolasbock** [comment]
@MaximilianMeister Thanks for the explanation. That works for me.

File: lib/crowbar/client/mixin/database.rb (hunk 1 above)
Comment (aplanas) [inline lib/crowbar/client/mixin/database.rb]: LGTM but a port can't start with 0?

Comment (nicolasbock) [inline lib/crowbar/client/mixin/database.rb]: @MaximilianMeister I am assuming the port is used in an integer context at some point, which would make leading zeros ok. Is that correct?

**Pull request merged**